tools/code_coverage.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/*
+ * This tool is designed to generate code coverage reports for the unit testing suite of LORIS.
+ *
+ * Please run this command from the tools directory -- var/www/loris/tools
+ *
+ * First, this tool will run all the unit tests on the particular branch. This will automatically
+ * generate HTML reports (configured in the phpunit.xml file). Then, the generated files will be
+ * moved to the directory "htdocs/log/codeCoverage/<date>", where the date value is the current date. Finally,
+ * all the HTML reports are edited to include the date they were generated at the top of the page.
+ *
+ * To view the generated HTML reports, go to https://<your-loris-url>/log/codeCoverage/<date>/index.html.
+ * This will take you to the main coverage page. From here, you may navigate to the dashboard or
+ * to the pages for the individual classes under the php/libraries directory.
+ *
+*/
+
+/*
+ * Step 1: Running unit test suite and generating HTML reports
+ *
+ * The unit test suite is run with the --coverage-html tag,
+ * which is configured to generate the HTML reports
+ */
+error_reporting(E_ERROR);
+exec(
+    "docker-compose run -T --rm unit-tests vendor/bin/phpunit --configuration test/phpunit.xml --testsuite LorisUnitTests --coverage-html htdocs/log/codeCoverage",
+    $output,
+    $status
+);
+if ($status != 0) {
+    echo PHP_EOL . "The unit tests did not execute correctly" . PHP_EOL;
+}
+else {
+    echo "The unit test suite passed with no errors" . PHP_EOL;
+}
+echo end($output) . PHP_EOL;
+
+/*
+ * Step 2: Change the read/write permissions of the HTML files
+ */
+exec("sudo chmod -R 777 ../htdocs/log/codeCoverage", $output, $status);
```

Comment:
This script should not be executing commands with sudo

Reply:
@driusan I did this because when the HTML files are generated, they are given root user permissions so I can't make any edits to them. I can try to do this another way though.

Reply:
Why are they generated as root? They should be owned by the person who created them. It might be because of the docker-compose which you've removed.

Reply:
So I did some more digging. When you setup docker, you can add your user (ex. lorisadmin) as part of the group so that you can run tests without `sudo`. BUT the Docker daemon still runs as a root user, so anything it generates will be generated with root permissions. There is a way to run the Docker daemon as a non-root user (instructions [here](https://docs.docker.com/engine/security/rootless/)), but it would involve changing all our setup instructions for setting up the docker environment.